From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Leo Sandoval <lsandova@redhat.com>
Date: Fri, 5 Apr 2024 10:55:39 -0600
Subject: [PATCH] post-fix: x86-efi: Make our own allocator for kernel stuff

---
 grub-core/loader/i386/efi/linux.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/grub-core/loader/i386/efi/linux.c b/grub-core/loader/i386/efi/linux.c
index 9e98892b3d0..039b40e321d 100644
--- a/grub-core/loader/i386/efi/linux.c
+++ b/grub-core/loader/i386/efi/linux.c
@@ -208,7 +208,7 @@ grub_cmd_initrd (grub_command_t cmd __attribute__ ((unused)),
   grub_dprintf ("linux", "initrd_mem = %p\n", initrd_mem);
 
   params->ramdisk_size = size;
-  params->ramdisk_image = initrd_mem;
+  params->ramdisk_image = (grub_uint64_t) initrd_mem;
 
   ptr = initrd_mem;
 
@@ -371,9 +371,8 @@ grub_cmd_linux (grub_command_t cmd __attribute__ ((unused)),
 			      GRUB_VERIFY_KERNEL_CMDLINE);
 
   grub_dprintf ("linux", "cmdline:%s\n", linux_cmdline);
-  grub_dprintf ("linux", "setting lh->cmd_line_ptr to 0x%08x\n",
-		linux_cmdline);
-  lh->cmd_line_ptr = linux_cmdline;
+
+  lh->cmd_line_ptr = (grub_uint64_t) linux_cmdline;
 
   handover_offset = lh->handover_offset;
   grub_dprintf("linux", "handover_offset: 0x%08x\n", handover_offset);
